source/tests/system/nisync_driver_api_tests.cpp

```diff
@@ -540,6 +557,36 @@ TEST_F(NiSyncDriverApiTest, AttributeSet_GetAttributeViString_ReturnsValue)
   EXPECT_STREQ(expectedValue, value.c_str());
 }
 
+TEST_F(NiSyncDriverApiTest, MeasureFrequencyExOnTerminalWithNoFrequency_ReturnsNoFrequency)
+{
+  ViStatus viStatus;
+  ViConstString srcTerminal = NISYNC_VAL_PFI0;
+  ViReal64 duration = 0.1; // duration is in seconds
+  ViUInt32 decimationCount = 0; // ignored for 6674
+  ViReal64 actualDuration, frequency, frequencyError;
+  auto grpcStatus = call_MeasureFrequencyEx(srcTerminal, duration, decimationCount, &actualDuration, &frequency, &frequencyError, &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+  EXPECT_EQ(0, actualDuration);
+  EXPECT_EQ(0, frequency);
+}
+
+TEST_F(NiSyncDriverApiTest, MeasureFrequencyExOnOscillatorWithFrequency_ReturnsFrequency)
+{
+  ViStatus viStatus;
+  ViConstString srcTerminal = NISYNC_VAL_OSCILLATOR;
+  ViReal64 duration = 0.1; // duration is in seconds
+  ViUInt32 decimationCount = 0; // ignored for 6674
+  ViReal64 actualDuration, frequency, frequencyError;
+  auto grpcStatus = call_MeasureFrequencyEx(srcTerminal, duration, decimationCount, &actualDuration, &frequency, &frequencyError, &viStatus);
+
+  EXPECT_TRUE(grpcStatus.ok());
+  EXPECT_EQ(VI_SUCCESS, viStatus);
+  EXPECT_GT(actualDuration, 0);  
```

Comment:
For both of these values I'm just checking that they are greater than zero.  In reality it should be really close to 0.1 for the duration and really close to 10 for the frequency.

Reply:
This seems fine to me. More of an FYI, but Google Test has some assertions for floats / doubles that assert the values are "close" to each other and one that takes in an error bound (see https://github.com/google/googletest/blob/master/docs/advanced.md#floating-point-comparison).

I expect just checking greater than zero is fine here but those are some alternative assertions you can use.

Reply:
For cases where you would like to test specific values or corner cases, you may want to consider adding an integration test that uses a mocked version of your service. Also the benefit in doing so is that the integration tests run on the CI while the system tests do not. This test looks good as is though.

Reply:
I really just wanted to do sanity testing here to make sure things look reasonable.  Thanks for the suggestions about floating point and the integration tests.  I'll keep that in mind for future changes.  I'm going to leave this as is for now.